src/data/set/intervals/unordered_interval.lean

```diff
@@ -124,6 +130,20 @@ begin
     { exact or.inl (mem_interval_of_ge hb hx.2) } }
 end
 
+lemma eq_of_mem_interval_of_mem_interval : a ∈ interval b c → b ∈ interval a c → a = b :=
```

Comment:
I guess the swapped version of this is worth having too, where the left argument to `interval` is fixed